### PR TITLE
Add snapps transactions to Transaction.t

### PR DESCRIPTION
**Pull request by imeckler**
This PR changes the transaction type and integrates snapp transactions into coda. It does not change the transaction pool to support them. That will be included in an upcoming PR.

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/coda_base/command_transaction.ml [hunk 1]
```diff
@@ -0,0 +1,229 @@
+open Core_kernel
```
Comment (bkase) [inline src/lib/coda_base/command_transaction.ml:1]: Introducing a new tier here above user-command is going to cause a lot of churn around archive-node, graphql, rosetta, cli, documentation and probably more.

Could we nest this under user-command? Or make it a sibling of user-command and fee-transfer and coinbase in trasnaction.ml instead?

**Review by bkase — CHANGES_REQUESTED** [review]
You still can't create Snapp transactions yet right? As soon as you can, we'll want to make sure the cli, graphql, rosetta, and docs are all updated atomically in the same PR.

File: src/lib/coda_base/command_transaction.ml (hunk 1 above)
Comment (imeckler) [reply src/lib/coda_base/command_transaction.ml:1]: I think the easiest thing at this point would be to rename Command_transaction to User_command and rename User_command to Signed_transaction? Would that be ok?

Comment (bkase) [reply src/lib/coda_base/command_transaction.ml:1]: Yes, this seems okay!

File: src/app/archive/archive_lib/processor.ml [hunk 2]
```diff
@@ -93,152 +108,191 @@ module Snarked_ledger_hash = struct
...
+          ( (Some status_str, failure_reason)
+          , ( fee_payer_account_creation_fee_paid
+            , receiver_account_creation_fee_paid
+            , created_token )
+          , user_command_id )
+      in
+      user_command_id
+  end
+
+  let as_user_command (t : User_command.t) : Coda_base.Signed_command.t =
```
Comment (bkase) [inline src/app/archive/archive_lib/processor.ml:260]: as_signed_command

File: src/app/archive/archive_lib/processor.ml [hunk 3]
```diff
@@ -93,152 +108,191 @@ module Snarked_ledger_hash = struct
...
+    match t with
+    | Signed_command c ->
+        c
+    | Snapp_command c ->
+        let module S = Coda_base.Snapp_command in
+        let ({source; receiver; amount} : S.transfer) = S.as_transfer c in
+        let fee_payer = S.fee_payer c in
+        { signature= Signature.dummy
```
Comment (bkase) [inline src/app/archive/archive_lib/processor.ml:268]: Can we still disambiguate between normal signed_command payments and snap commands coerced into payments via the `typ` column in the db? Could you put some metadata here in order to recover that?

**Review by bkase — APPROVED** [review]

**Pull request merged**